### PR TITLE
develop: update the third party support table

### PR DESCRIPTION
**Pull request by enkilee**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [x] v7.4 (TiDB 7.4 versions)
- [x] v7.3 (TiDB 7.3 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/15187
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-third-party-support.md [hunk 1]
```diff
@@ -140,30 +140,37 @@ PingCAP 与开源社区合作，通过三方工具提供以下支持：
          <td>v7.0</td>
          <td>Full</td>
          <td>N/A</td>
-         <td>N/A</td>
+         <td><a href="/tidb/dev/dev-guide-sample-application-ruby-rails">Connect to TiDB with Rails Framework and ActiveRecord ORM</a></td>
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-support.md:143]:
```suggestion
         <td><a href="/zh/tidb/dev/dev-guide-sample-application-ruby-rails">使用 Rails 框架和 ActiveRecord ORM 连接 TiDB</a></td>
```

Comment (Oreoxmt) [reply develop/dev-guide-third-party-support.md:143]: @enkilee 非常感谢您对 TiDB 文档的贡献！

对于中文文档，请使用中文标题，并在 href 中添加 `/zh` 前缀。

File: develop/dev-guide-third-party-support.md [hunk 2]
```diff
@@ -140,30 +140,37 @@ PingCAP 与开源社区合作，通过三方工具提供以下支持：
...
       </tr>
       <tr>
          <td rowspan="2">Python</td>
          <td><a href="https://pypi.org/project/Django/" target="_blank" referrerpolicy="no-referrer-when-downgrade">Django</a></td>
-         <td>v4.1</td>
+         <td>v4.2</td>
          <td>Full</td>
          <td><a href="https://github.com/pingcap/django-tidb" target="_blank" referrerpolicy="no-referrer-when-downgrade">django-tidb</a></td>
-         <td><a href="/zh/tidb/dev/dev-guide-sample-application-django">使用 Django 连接到 TiDB</a></td>
+         <td><a href="/tidb/dev/dev-guide-sample-application-python-django">Connect to TiDB with Django</a></td>
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-support.md:173]:
```suggestion
         <td><a href="/zh/tidb/dev/dev-guide-sample-application-django">使用 Django 连接到 TiDB</a></td>
```

Comment (enkilee) [reply develop/dev-guide-third-party-support.md:173]: 收到，抱歉。已改

File: develop/dev-guide-third-party-support.md (hunk 1 above)
Comment (enkilee) [reply develop/dev-guide-third-party-support.md:143]: 收到，抱歉。已改

File: develop/dev-guide-third-party-support.md [hunk 3]
```diff
@@ -177,6 +184,8 @@ PingCAP 与开源社区合作，通过三方工具提供以下支持：
 
 ## GUI
 
-| GUI | 最新已测试版本 | 支持等级 | 教程 |
-| - | - | - | - |
-| [DBeaver](https://dbeaver.io/) | 23.0.1 | Full | N/A |
+| GUI                                                       | Latest tested version | Support level | Tutorial                                                                      |
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-support.md:187]: GUI 部分请翻译为中文

Comment (enkilee) [reply develop/dev-guide-third-party-support.md:187]: 收到，已改

File: develop/dev-guide-third-party-support.md [hunk 4]
```diff
@@ -140,30 +140,37 @@ PingCAP 与开源社区合作，通过三方工具提供以下支持：
...
       <tr>
-         <td rowspan="2">JavaScript / TypeScript</td>
-         <td><a href="https://www.npmjs.com/package/sequelize" target="_blank" referrerpolicy="no-referrer-when-downgrade">sequelize</a></td>
+         <td rowspan="3">JavaScript / TypeScript</td>
+         <td><a href="https://sequelize.org/" target="_blank" referrerpolicy="no-referrer-when-downgrade">Sequelize</a></td>
          <td>v6.20.1</td>
          <td>Full</td>
          <td>N/A</td>
-         <td>N/A</td>
+         <td><a href="/tidb/dev/dev-guide-sample-application-nodejs-sequelize">Connect to TiDB with Sequelize</a></td>
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-support.md:151]:
```suggestion
         <td>N/A</td>
```

File: develop/dev-guide-third-party-support.md (hunk 3 above)
Comment (Oreoxmt) [inline develop/dev-guide-third-party-support.md:187]:
```suggestion
| GUI                                                       | 最新已测试版本 | 支持等级 | 教程                                                                      |
```

File: develop/dev-guide-third-party-support.md [hunk 5]
```diff
@@ -177,6 +184,8 @@ PingCAP 与开源社区合作，通过三方工具提供以下支持：
...
 ## GUI
 
-| GUI | 最新已测试版本 | 支持等级 | 教程 |
-| - | - | - | - |
-| [DBeaver](https://dbeaver.io/) | 23.0.1 | Full | N/A |
+| GUI                                                       | Latest tested version | Support level | Tutorial                                                                      |
+|-----------------------------------------------------------|-----------------------|---------------|-------------------------------------------------------------------------------|
+| [JetBrains DataGrip](https://www.jetbrains.com/datagrip/) | 2023.2.1              | Full          | [使用 JetBrains DataGrip 连接到 TiDB](/develop/dev-guide-gui-datagrip.md) |
+| [DBeaver](https://dbeaver.io/)                            | 23.0.3                | Full          | [使用 DBeaver 连接到 TiDB](/develop/dev-guide-gui-dbeaver.md)             |
+| [Visual Studio Code](https://code.visualstudio.com/)                            | 1.72.0                | Full          | [使用 Visual Studio Code 连接到 TiDB](/develop/dev-guide-gui-vscode-sqltools.md)             |
```
Comment (Oreoxmt) [inline develop/dev-guide-third-party-support.md:191]:
```suggestion
| [JetBrains DataGrip](https://www.jetbrains.com/datagrip/) | 2023.2.1              | Full          | N/A |
| [DBeaver](https://dbeaver.io/)                            | 23.0.3                | Full          | N/A             |
| [Visual Studio Code](https://code.visualstudio.com/)                            | 1.72.0                | Full          | N/A             |
```
因为这些文档目前没有中文版，所以这里教程列修改为 N/A

Comment (enkilee) [reply develop/dev-guide-third-party-support.md]: 收到

File: develop/dev-guide-third-party-support.md (hunk 3 above)
Comment (enkilee) [reply develop/dev-guide-third-party-support.md]: 收到

File: develop/dev-guide-third-party-support.md (hunk 4 above)
Comment (enkilee) [reply develop/dev-guide-third-party-support.md]: 收到

**Review by Mini256 — APPROVED** [review]
LGTM, thanks a lot!

**Review by Oreoxmt — APPROVED** [review]

**Pull request merged**